### DISCUSSION TITLE
Improve caching

### DISCUSSION THREAD

**Discussion by thewh1teagle** [Mozilla-Actions/sccache-action#158] [General]
Hi
I setup sccache for c++ project (cross platform)
but seems like it's not caching correctly.

https://github.com/thewh1teagle/loud.cpp/pull/16
https://github.com/thewh1teagle/loud.cpp/actions/runs/11821346721/job/32935862517

The compile time remain similar as before the caching
how can I improve it?
Thanks

![image](https://github.com/user-attachments/assets/d59a1368-5c62-4911-8f81-e49b65b9acb6)

**Comment by sylvestre** [comment]
Probably because the second build doesn't generate the same hash
see the doc for the list:
https://github.com/mozilla/sccache/blob/main/docs/Caching.md

**Reply by thewh1teagle** [reply]
It means that it doesn't restore the right cache? because as you can see, the problem seems to be that the content of the cache is too small. it caches only few KB. the program is much larger with shared libs and more.

Also I run the same action it should be identical

Maybe I miss understood something, does it also cache things that cmake fetch with fetchContentDeclare and similar?

**Reply by sylvestre** [reply]
yes, you should build with debug log to identify the hash.

Note that the path is part of the hash

I don't think the size matters. it is still faster to download than compiling
